crates/eth-rpc/src/api/eth_api.rs

```diff
@@ -211,4 +211,28 @@ pub trait EthApi {
         keys: Vec<H256>,
         block_id: Option<BlockId>,
     ) -> Result<EIP1186AccountProofResponse>;
+
+    /// Creates a filter object, based on filter options, to notify when the state changes (logs).
+    #[method(name = "newFilter")]
+    async fn new_filter(&self, filter: Filter) -> Result<U64>;
+
+    /// Creates a filter in the node, to notify when a new block arrives.
+    #[method(name = "newBlockFilter")]
+    async fn new_block_filter(&self) -> Result<U64>;
+
+    /// Creates a filter in the node, to notify when new pending transactions arrive.
+    #[method(name = "newPendingTransactionFilter")]
+    async fn new_pending_transaction_filter(&self) -> Result<U64>;
+
+    /// Destroys a filter based on filter ID
+    #[method(name = "uninstallFilter")]
+    async fn uninstall_filter(&self, id: U64) -> Result<bool>;
+
+    /// Returns a list of all logs based on filter ID since the last log retrieval
+    #[method(name = "getFilterChanges")]
+    async fn get_filter_changes(&self, id: U64) -> Result<Vec<Log>>;
```

Comment:
You are not using the correct return type for the function, please have a look here:
https://github.com/ethereum/execution-apis/blob/main/src/schemas/filter.yaml#L1

Same for getFilterLogs.

Please note [here](https://github.com/paradigmxyz/reth/blob/main/crates/rpc/rpc-types/src/eth/filter.rs#L821) the type used by Reth.

Reply:
Resolved!